src/plugins/intel_npu/src/plugin/src/plugin.cpp

```diff
@@ -471,7 +471,7 @@ std::shared_ptr<ov::ICompiledModel> Plugin::compile_model(const std::shared_ptr<
     if (!set_cache_dir.empty()) {
         const auto compilerType = localConfig.get<COMPILER_TYPE>();
         if (compilerType == ov::intel_npu::CompilerType::MLIR) {
-            _logger.error("Option 'CACHE_DIR' is not supported with MLIR compiler type");
+            OPENVINO_THROW("Option 'CACHE_DIR' is not supported with MLIR compiler type");
```

Comment:
do we need to change it?

Reply:
@sbutnari  Pls help clarify.

Reply:
This 'if' branch is triggered when the user wants to compile a blob with MLIR, they're likely expecting a change in their blob.
The change is acceptable, because it lets a user know that they'll be using an outdated blob. Better throw an error than log it silently.